### ISSUE TITLE
Mismatched or dirty k/release does not block r

### ISSUE THREAD

**Issue by justaugustus** [kubernetes/release#1278]
#### What happened:

The repo state check in https://github.com/kubernetes/release/pull/1254 does not appear to block gcbmgr runs with a dirty or out of sync local copy of k/release.

/assign @saschagrunert

**Comment by saschagrunert** [comment]
Yes, my template for the implementation was this function, which do not check those cases as well: 
https://github.com/kubernetes/release/blob/61c27116a97935a592fcb58fdd4f115bfd3a563a/lib/gitlib.sh#L296-L328

So we can add two possible checks:
- Error if the repository is dirty
- Error if the current commit does not match exactly the latest remote commit

In case of the second error path the `branch --contains` check could be replaces with a basic SHA comparison. WDYT? Should we add both?

**Comment by justaugustus** [comment]
@saschagrunert -- Let's aim to check for both. The shell-based `gcbmgr` did check for the latter.

**Comment by saschagrunert** [comment]
Fix is incoming in https://github.com/kubernetes/release/pull/1284
